pinot-controller/src/main/java/org/apache/pinot/controller/util/ServerSegmentMetadataReader.java

```diff
@@ -68,7 +68,7 @@ public class ServerSegmentMetadataReader {
 
   public ServerSegmentMetadataReader() {
     _executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
-    _connectionManager = new PoolingHttpClientConnectionManager();
+    _connectionManager = PoolingHttpClientConnectionManagerHelper.createWithSocketFactory();
```

Comment:
How is this ensuring we are connecting over HTTPS and not HTTP ?

Reply:
This step does not ensure that. It just configures the SSLSocketFactory to use the keystore (and truststore) that is configured in the Pinot configuration. Currently this one picks up the system default.